File: clients/rust/model/src/unit.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct Unit {
    pub player_id: i32,
    pub id: i32,
    pub health: i32,
    pub position: Vec2F64,
    pub size: Vec2F64,
    pub jump_state: JumpState,
    pub walked_right: bool,
    pub stand: bool,
    pub on_ground: bool,
    pub on_ladder: bool,
    pub mines: i32,
    pub weapon: Option<Weapon>,
}

File: clients/haskell/src/RAIC/Model/Unit.hs
module RAIC.Model.Unit where

import           GHC.Generics          (Generic)
import           RAIC.Model.JumpState  (JumpState)
import           RAIC.Model.Vec2Double (Vec2Double)
import           RAIC.Model.Weapon     (Weapon)
import           RAIC.Utils.Trans      (Trans)

data Unit = Unit {
  player_id    :: Int,
  id           :: Int,
  health       :: Int,
  position     :: Vec2Double,
  size         :: Vec2Double,
  jump_state   :: JumpState,
  walked_right :: Bool,
  stand        :: Bool,
  on_ground    :: Bool,
  on_ladder    :: Bool,
  mines        :: Int,
  weapon       :: Maybe Weapon
} deriving (Generic, Show)

instance Trans Unit
